modules/shredder/src/main/scala/com/snowplowanalytics/snowplow/shredder/ShredJob.scala

```diff
@@ -133,34 +133,30 @@ class ShredJob(@transient val spark: SparkSession, shredConfig: CliConfig) exten
         timestampsAccumulator.add(event)
         val isSyntheticDupe = syntheticDupesBroadcasted.value.contains(event.event_id)
         val withDupeContext = if (isSyntheticDupe) Deduplication.withSynthetic(event) else event
-        FinalRow.shred(shredConfig.igluConfig, isTabular, atomicLengths)(withDupeContext)
+        Shredded.fromEvent(shredConfig.igluConfig, isTabular, atomicLengths)(withDupeContext)
       }
     }.cache()
 
-    val shreddedGood = shredded.flatMap(_.toOption)
-
-    // Ready the events for database load
-    val events = shreddedGood.map(_.atomic)
-
     // Update the shredded JSONs with the new deduplicated event IDs and stringify
-    val shreddedData = shreddedGood.flatMap(_.shredded)
+    val shreddedData = shredded.flatMap(_.getOrElse(Nil))
 
     // Data that failed TSV transformation
     val shreddedBad = (common.flatMap(_.swap.toOption) ++ shredded.flatMap(_.swap.toOption)).map(bad => Row(bad.compact))
 
-    // Write as strings to `atomic-events` directory
-    Sink.writeEvents(spark, shredder.compression, events, outFolder)
```

Comment:
One Spark job removed, that's awesome !!

I'm wondering if we could consider bad rows in the same way as atomic and shredded data to have one job less

Reply:
I think we can. But that would mean we need to drop configurable `badOutput` path. For example we can do following partitioning:

```
run=2021-01-07-20-35-30/kind=good/vendor=com.acme/name=event/format=tsv/model=1
run=2021-01-07-20-35-30/kind=bad/vendor=com.snowplowanalytics/name=shredding-error/format=json/model=1
```

...but is it bad actually? I think it's a great idea and we should do that. Just given the time pressure not for this release maybe - I'll fill a ticket.